drivers/network/tcpip/datalink/lan.c

```diff
@@ -750,54 +750,47 @@ BOOLEAN ReconfigureAdapter(PRECONFIGURE_CONTEXT Context)
 
     Context->Adapter->CompletingReset = FALSE;
 
-    /* Update the IP and link status information cached in TCP */
-    TCPUpdateInterfaceIPInformation(Interface);
-    TCPUpdateInterfaceLinkStatus(Interface);
-
-    /* We're done here if the adapter isn't connected */
-    if (Context->State != LAN_STATE_STARTED)
+    if (Context->State == LAN_STATE_STARTED)
     {
-        Adapter->State = Context->State;
-        return TRUE;
-    }
-
-    /* NDIS Bug! */
-#if 0
-    /* Get maximum link speed */
-    NdisStatus = NDISCall(Adapter,
-                          NdisRequestQueryInformation,
-                          OID_GEN_LINK_SPEED,
-                          &Interface->Speed,
-                          sizeof(UINT));
-
-    if (!NT_SUCCESS(NdisStatus))
-        Interface->Speed = IP_DEFAULT_LINK_SPEED;
-
-    Adapter->Speed = Interface->Speed * 100L;
-
-    /* Get maximum frame size */
-    NdisStatus = NDISCall(Adapter,
-                          NdisRequestQueryInformation,
-                          OID_GEN_MAXIMUM_FRAME_SIZE,
-                          &Adapter->MTU,
-                          sizeof(UINT));
-    if (NdisStatus != NDIS_STATUS_SUCCESS)
-        return FALSE;
+        /* Get maximum link speed */
+        NdisStatus = NDISCall(Adapter,
+                              NdisRequestQueryInformation,
+                              OID_GEN_LINK_SPEED,
+                              &Interface->Speed,
+                              sizeof(UINT));
```

Comment:
Suggestion, since this is mostly about improving the original (but previously disabled!) code:
Maybe `sizeof(Interface->Speed)` and similar for the other ones.

Reply:
Done